src/main.rs

```diff
@@ -73,15 +74,38 @@ fn remove(opt: cli::Opt, key: String) -> Result<()> {
     config.save(&opt.amber_yaml)
 }
 
-fn print(opt: cli::Opt) -> Result<()> {
+fn print(opt: cli::Opt, style: cli::PrintStyle) -> Result<()> {
     let config = config::Config::load(&opt.amber_yaml)?;
     let secret = config.load_secret_key()?;
     let pairs: Result<Vec<_>> = config.iter_secrets(&secret).collect();
     let mut pairs = pairs?;
     pairs.sort_by(|x, y| x.0.cmp(y.0));
-    pairs
-        .iter()
-        .for_each(|(key, value)| println!("export {}={:?}", key, value));
+
+    let to_objs = |p: Vec<(&String, String)>| {
+        p.iter()
+            .map(|(k, v)| {
+                let mut obj = HashMap::new();
+                obj.insert("key", (*k).clone());
+                obj.insert("value", v.clone());
+                obj
+            })
+            .collect::<Vec<_>>()
+    };
+    match style {
+        cli::PrintStyle::SetEnv => pairs
+            .iter()
+            .for_each(|(key, value)| println!("export {}={:?}", key, value)),
+        cli::PrintStyle::Json => {
+            let secrets = to_objs(pairs);
+            let json = serde_json::to_string(&secrets).unwrap();
```

Comment:
Slight optimization: you could use `to_writer` to directly print to `stdout` instead of going through a `String`. Also, as a matter of principle, I'd rather avoid the `.unwrap()`, though I admit it's probably harmless in this case.

Reply:
My bad, forgot the function return type is Result.